WeakAuras/TSUHelpers.lua

```diff
@@ -0,0 +1,85 @@
+if not WeakAuras.IsLibsOK() then return end
+---@type string
+local AddonName = ...
+---@class Private
+local Private = select(2, ...)
+
+---@alias key string | integer
+---@alias states table<key, state>
+
+---@type fun(state: state)
+local function fixMissingFields(state)
+  if type(state) ~= "table" then return end
+  -- set show
+  if state.show == nil then
+    state.show = true
+  end
+end
+
+---@type fun(states: states, key: key): boolean
+local remove = function(states, key)
+  local changed = false
+  local state = states[key]
+  if state then
+    state.show = false
+    state.changed = true
+    changed = true
+  end
+  return changed
+end
+
+---@type fun(states: states): boolean
+local removeAll = function(states)
+  local changed = false
+  for _, state in pairs(states) do
+    state.show = false
+    state.changed = true
+    changed = true
+  end
+  return changed
+end
+
+---@type fun(states: states, newState: state, key: key): boolean
```

Comment:
That feels like the wrong order to me, "key, state" feels a lot more natural to me.

I know you did that for key to be optional, but I still feel that its the wrong order.

Reply:
swapped them, and made update function check differences recursively (eg. for overlays)